cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -709,6 +709,41 @@ impl ManagedEnvironment {
 
 /// Utility instance methods
 impl ManagedEnvironment {
+    /// Edit the environment without checking that it builds
+    ///
+    /// This is used to allow `flox pull` to work with environments
+    /// that don't specify the current system as supported.
+    pub fn edit_unsafe(
+        &mut self,
+        flox: &Flox,
+        contents: String,
+    ) -> Result<EditResult, EnvironmentError2> {
+        let mut generations = self
+            .generations()
+            .writable(flox.temp_dir.clone())
+            .map_err(ManagedEnvironmentError::CreateFloxmetaDir)?;
+        let mut temporary = generations
+            .get_current_generation()
+            .map_err(ManagedEnvironmentError::CreateGenerationFiles)?;
+
+        let result = temporary.edit_unsafe(flox, contents)?;
+
+        if result == EditResult::Unchanged {
+            return Ok(result);
+        }
+
+        debug!("Environment changed, create generation, lock generation, build and link");
```

Comment:
nit: I don't understand this message. Do you mean `attempting to create...`? Also it shouldn't say link